File: javascript/6.js
const sumSquares = (n) => {
	return [...Array(n+1).keys()]
				.map((n) => Math.pow(n, 2))
				.reduce((x, y) => { return x + y; });
};

const squareSums = (n) => {
	return Math.pow([...Array(n+1).keys()]
				.reduce((x, y) => { return x + y; }), 2);
};

const diff = (n) => {
	return squareSums(n) - sumSquares(n);
}

// Difference between sum of squares of the first one hundred natural numbers and the square of the sum
console.log(diff(100));

File: python/6.py
def sum_of_squares(n):
	return sum([i**2 for i in xrange(n+1)])

def square_of_sums(n):
	return sum([i for i in xrange(n+1)])**2

def diff(n):
	return square_of_sums(n) - sum_of_squares(n)

# Difference between sum of squares of the first one hundred natural numbers and the square of the sum
print diff(100)
